tests/reftests/repository.test

```diff
@@ -316,8 +316,13 @@ first  --
 ### git -C ./OPER add repo
 ### git -C ./OPER add packages/first
 ### git -C ./OPER commit -qm "third"
+### git -C ./OPER commit --allow-empty -m "second empty commit" --quiet
```

Comment:
this is no more needed too, as everything is seen in `source` test

Reply:
This was commented by @kit-ty-kate. It's about avoiding regression in the future. Maybe we could refactor it like source.

Reply:
You're right.

Reply:
Done in https://github.com/ocaml/opam/pull/5888/commits/29bc4a513a95fdca3ca00d1c42704eb6f0344bbb.

Reply:
why is it not needed? I'm not seeing that test in `source.test` (which also isn't really the place to test for this if it was there)

Reply:
It is having `--dir` which use "--depth 1" and  `--dev` which doesn't.  I think this is what @rjbou meant.
```
### opam source pandore.4 --dir pandore5
Successfully extracted to ${BASEDIR}/pandore5
### git -C pandore5 rev-list --all --count
1
### opam source pandore.4 --dev --dir pandore6
Successfully fetched pandore development repo to ${BASEDIR}/pandore6
### git -C pandore6 rev-list --all --count
2
```

Reply:
> It is having `--dir` which use "--depth 1"

what is the reasoning for this?

Reply:
oh you mean without `--dev` yeah sure. Though it's not testing the same thing. Repositories don't use the same code path.

Reply:
I've just rephrased my comment.

Reply:
> oh you mean without `--dev` yeah sure. Though it's not testing the same thing. Repositories don't use the same code path.

 Ah, It's what I meant.

Reply:
How do you feel adding, the meaning line that I removed inside repository.test instead ?
```
### git -C ./OPAM/repo/oper rev-list --all --count
1
```

Reply:
I think 29bc4a513a95fdca3ca00d1c42704eb6f0344bbb should be reverted

Reply:
You're right, I've just realized that I'm mixing everything.

Reply:
@kit-ty-kate what is the point of this test ? I'm guessing that it is to be sure that opam repo download is kept with `--depth 1`, is it? Its code path remain untouched, but indeed it calls `VCS.fetch`.

Reply:
yeah exactly. So that if in the future the behaviour changes (by mistake or on purpose) we already have a test for it

Reply:
Ok! It needs then a comment to explain why it is here. I'll add it.